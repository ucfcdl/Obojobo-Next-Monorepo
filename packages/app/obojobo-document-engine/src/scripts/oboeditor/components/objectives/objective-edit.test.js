import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import ObjectiveEdit from 'src/scripts/oboeditor/components/objectives/objective-edit'

describe('Objective Edit', () => {
	test('ObjectiveEdit renders correctly', () => {
		const props = {
			data: {
				id: 'mock-id',
				label: 'mock-label',
				description: 'description'
			},
			onCancel: jest.fn(),
			onConfirm: jest.fn()
		}

		const component = renderer.create(<ObjectiveEdit {...props} />)
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('ObjectiveEdit allows confirmation', () => {
		const props = {
			data: {
				id: 'mock-id',
				label: 'mock-label',
				description: 'description'
			},
			onCancel: jest.fn(),
			onConfirm: jest.fn()
		}

		const component = mount(<ObjectiveEdit {...props} />)

		// Writing mock label and description
		component.find('input#objective-label').simulate('change', {
			target: { value: 'mock-label' }
		})

		component.find('input#objective-input').simulate('change', {
			target: { value: 'mock-description' }
		})

		// Actual testing starts here
		component
			.find('button')
			.at(1)
			.simulate('click')
		expect(props.onConfirm).toHaveBeenCalled()
	})

	test('ObjectiveEdit does not allow confirmation if label and decription are missing', () => {
		const props = {
			data: {
				id: 'mock-id',
				label: '',
				description: ''
			},
			onCancel: jest.fn(),
			onConfirm: jest.fn()
		}

		const component = mount(<ObjectiveEdit {...props} />)

		component
			.find('button')
			.at(1)
			.simulate('click')
		expect(props.onConfirm).not.toHaveBeenCalled()
	})

	test('ObjectiveEdit does not allow confirmation if label is missing', () => {
		const props = {
			data: {
				id: 'mock-id',
				label: '',
				description: ''
			},
			onCancel: jest.fn(),
			onConfirm: jest.fn()
		}

		const component = mount(<ObjectiveEdit {...props} />)

		// Writing mock description
		component.find('input#objective-input').simulate('change', {
			target: { value: 'mock-description' }
		})

		// Actual testing starts here
		component
			.find('button')
			.at(1)
			.simulate('click')
		expect(props.onConfirm).not.toHaveBeenCalled()
	})

	test('ObjectiveEdit does not allow confirmation if description is missing', () => {
		const props = {
			data: {
				id: 'mock-id',
				label: '',
				description: ''
			},
			onCancel: jest.fn(),
			onConfirm: jest.fn()
		}

		const component = mount(<ObjectiveEdit {...props} />)

		// Writing mock label
		component.find('input#objective-label').simulate('change', {
			target: { value: 'mock-label' }
		})

		// Actual testing starts here
		component
			.find('button')
			.at(1)
			.simulate('click')
		expect(props.onConfirm).not.toHaveBeenCalled()
	})
})
