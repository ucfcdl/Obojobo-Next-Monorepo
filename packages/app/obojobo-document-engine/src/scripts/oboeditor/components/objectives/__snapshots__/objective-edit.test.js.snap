// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Objective Edit ObjectiveEdit renders correctly 1`] = `
<div
  className="obojobo-draft--components--modal--simple-dialog"
>
  <div
    className="obojobo-draft--components--modal--dialog "
    style={null}
  >
    <div
      aria-atomic="true"
      aria-labelledby="obojobo-draft--components--modal--modal--content"
      aria-live="polite"
      className="obojobo-draft--components--modal--modal"
      role="dialog"
      tabIndex="-1"
    >
      <input
        aria-label="Start of dialog"
        className="first-tab"
        onFocus={[Function]}
        type="text"
      />
      <div
        className="content"
        id="obojobo-draft--components--modal--modal--content"
      >
        <h1
          className="heading"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Edit Objective
        </h1>
        <div
          className="dialog-content"
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <div
            className="objective-container"
          >
            <div>
              <label
                htmlFor="objective-label"
                style={
                  Object {
                    "marginRight": "1.5em",
                  }
                }
              >
                Label:
              </label>
              <input
                className="objective-input"
                id="objective-label"
                onChange={[Function]}
                placeholder="This objective's label (1.2, A, B, etc...) "
                type="text"
                value="mock-label"
              />
            </div>
            <div>
              <label
                htmlFor="objective-input"
                style={
                  Object {
                    "marginRight": "1.5em",
                  }
                }
              >
                Objective:
              </label>
              <input
                className="objective-input--objective"
                id="objective-input"
                onChange={[Function]}
                placeholder="Type your objective here..."
                type="text"
                value="description"
              />
            </div>
            <span
              className="objective-error"
            >
              
            </span>
          </div>
        </div>
        <div
          className="controls"
        >
          <div
            className="obojobo-draft--components--button alt-action is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[MockFunction]}
              type="button"
            >
              Cancel
            </button>
          </div>
          <div
            className="obojobo-draft--components--button is-not-dangerous align-center"
            contentEditable={false}
          >
            <button
              className="button"
              contentEditable={false}
              onClick={[Function]}
              type="button"
            >
              OK
            </button>
          </div>
        </div>
      </div>
      <input
        aria-label="End of dialog"
        className="last-tab"
        onFocus={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;
